src/Layer.js

```diff
@@ -147,29 +124,51 @@ export default class Layer {
      * @api
      */
     addTo (map, beforeLayerID) {
-        if (this._isCartoMap(map)) {
-            this._addToCartoMap(map, beforeLayerID);
-        } else if (this._isMGLMap(map)) {
-            this._addToMGLMap(map, beforeLayerID);
-        } else {
-            throw new CartoValidationError('layer', 'nonValidMap');
+        const STYLE_ERROR_REGEX = /Style is not done loading/;
+
+        try {
+            map.addLayer(this, beforeLayerID);
+        } catch (error) {
+            if (!STYLE_ERROR_REGEX.test(error)) {
+                throw new Error(error);
+            }
+
+            map.on('load', () => {
+                map.addLayer(this, beforeLayerID);
+            });
         }
     }
 
+    /**
+     * Remove this layer from a map.
+     *
+     * @param {mapboxgl.Map} map - The map on which to remove the layer
+     * @memberof carto.Layer
+     * @instance
+     * @api
+     */
+    removeFrom (map) {
+        map.removeLayer(this);
```

Comment:
This will throw an error if the map hasn't been loaded yet and the layer hasn't been added yet

Reply:
Yes and MGL this is the expected behaviour IMO.

Reply:
Yes, but since we are making things easier in addTo for a similar problem, we could improve this, but I realize that this feature request is not simple, I think we could create an issue with it, are you ok?